### ISSUE TITLE
feature: Visually organize complex flows s.t. less overlaps happen or overlaps are less confusing

### ISSUE THREAD

**Issue by wasnertobias** [windmill-labs/windmill#3741]
When there are a lot of branches and parallelism in a complex flow, it is tough to follow the lines of a branch, as any other elements overlap them. In addition, overlapping lines (crossings) make it hard to follow in a quick look.

Generally, I would love to see it less dense but more structured.

**Comment by wasnertobias** [comment]
Another idea: If there is a for loop or branch (if), then instead of indenting it as code, it could be visually indented as well, e.g., through shadows (higher z-axis)

One (not most extreme) example:
![Screenshot 2024-05-17 at 10 17 57](https://github.com/windmill-labs/windmill/assets/1354793/cdba1451-7fd8-42aa-b1ff-e20fd922b92f)

**Comment by wasnertobias** [comment]
Some "minimum example":
```
summary: ""
value:
  modules:
    - id: a
      value:
        type: branchall
        branches:
          - modules:
              - id: d
                value:
                  type: identity
                  flow: false
              - id: e
                value:
                  type: whileloopflow
                  modules:
                    - id: f
                      value:
                        type: identity
                        flow: false
                  skip_failures: false
            skip_failure: false
          - summary: ""
            expr: "false"
            modules:
              - id: b
                value:
                  type: forloopflow
                  modules:
                    - id: c
                      value:
                        type: identity
                        flow: false
                  iterator:
                    type: javascript
                    expr: ""
                  skip_failures: true
                  parallel: false
            skip_failure: false
          - summary: ""
            expr: "false"
            modules:
              - id: g
                value:
                  type: forloopflow
                  modules:
                    - id: i
                      value:
                        type: identity
                        flow: false
                  iterator:
                    type: javascript
                    expr: ""
                  skip_failures: true
                  parallel: false
            skip_failure: false
          - summary: ""
            expr: "false"
            modules: []
            skip_failure: false
        parallel: false
      summary: ""
    - id: h
      value:
        type: branchone
        branches:
          - summary: The default braunch looks like it goes to step n
            expr: "false"
            modules: []
          - summary: ""
            expr: "false"
            modules: []
          - summary: ""
            expr: "false"
            modules:
              - id: j
                value:
                  type: forloopflow
                  modules:
                    - id: l
                      value:
                        type: identity
                        flow: false
                  iterator:
                    type: javascript
                    expr: ""
                  skip_failures: true
                  parallel: false
              - id: m
                value:
                  type: branchall
                  branches:
                    - summary: ""
                      expr: "false"
                      modules:
                        - id: n
                          value:
                            type: identity
                            flow: false
                      skip_failure: false
                    - summary: ""
                      expr: "false"
                      modules: []
                      skip_failure: false
                  parallel: false
                summary: ""
              - id: k
                value:
                  type: identity
                  flow: false
          - summary: ""
            expr: "false"
            modules: []
          - summary: ""
            expr: "false"
            modules: []
          - summary: ""
            expr: "false"
            modules: []
        default: []
      summary: ""
schema:
  $schema: https://json-schema.org/draft/2020-12/schema
  properties: {}
  required: []
  type: object

```

**Comment by wasnertobias** [comment]
![Screenshot 2024-05-17 at 10 24 16](https://github.com/windmill-labs/windmill/assets/1354793/54b082ae-c695-4a0c-9cda-180039f3ce03)

**Comment by wasnertobias** [comment]
https://www.loom.com/share/011711e253614a7081a5dbaaf2cba5c6

**Comment by fatonramadani** [comment]
Hi we did some improvements recently, and we will on work on two special cases we have identified soon.

<img width="859" alt="Screenshot 2024-05-23 at 15 59 42" src="https://github.com/windmill-labs/windmill/assets/456655/11147c44-9628-4aed-85a0-54239996dff2">
